### PR TITLE
added on delete cascade to postgres

### PR DESCRIPTION
**Pull request by prithvi081099**

### CODE REVIEW

File: gateway/model/schema_type.go [hunk 1]
```diff
@@ -35,11 +35,11 @@ type (
 
 	// TableProperties are properties of the table
 	TableProperties struct {
-		From, To     string
-		Table, Field string
-		DBType       string
-		Group, Sort  string
-		Order        int
+		From, To                string
+		Table, Field, OnCascade string
```
Comment (YourTechBud) [inline gateway/model/schema_type.go]:
```suggestion
		Table, Field, OnDelete string
```

File: gateway/modules/schema/creation_test.go [hunk 2]
```diff
@@ -1557,6 +1557,19 @@ func TestSchema_generateCreationQueries(t *testing.T) {
 			want:    []string{"CREATE TABLE test.table1 (col2 timestamp );", "ALTER TABLE test.table1 ADD CONSTRAINT c_table1_col2 FOREIGN KEY (col2) REFERENCES test.table2 (id)"},
 			wantErr: false,
 		},
+		{
+			name: "adding a table and a foreign key with ON CASCADE DELETE",
```
Comment (YourTechBud) [inline gateway/modules/schema/creation_test.go]: Add the following test cases
- add a table along with foreign key with on cascade delete (you already have this
- Normal foreign key to foreign key with cascade delete
- Opposite of the above

File: gateway/modules/schema/helpers.go [hunk 3]
```diff
@@ -357,6 +360,10 @@ func (c *creationModule) modifyColumn() []string {
 	if c.realColumnInfo.IsForeign && !c.currentColumnInfo.IsForeign {
```
Comment (YourTechBud) [inline gateway/modules/schema/helpers.go]: What if both are `true` but the `OnCascade` parameter has changed?

File: gateway/modules/schema/schema.go [hunk 4]
```diff
@@ -236,6 +237,13 @@ func getCollectionSchema(doc *ast.Document, dbName, collectionName string) (mode
 							case "field", "to":
 								val, _ := utils.ParseGraphqlValue(arg.Value, nil)
 								fieldTypeStuct.JointTable.To = val.(string)
+
+							case "oncascade":
```
Comment (YourTechBud) [inline gateway/modules/schema/schema.go]:
```suggestion
							case "onDelete":
```

File: gateway/modules/schema/template.go [hunk 5]
```diff
@@ -9,7 +9,7 @@ import (
...
+	{{$fieldName}}: {{if eq $fieldValue.Kind "Object"}}{{$fieldValue.JointTable.Table}}{{else}}{{$fieldValue.Kind}}{{end}}{{if $fieldValue.IsFieldTypeRequired}}!{{end}} {{if $fieldValue.IsPrimary}}@primary{{end}} {{if $fieldValue.IsUnique}}@unique(group: "{{$fieldValue.IndexInfo.Group}}", order: {{$fieldValue.IndexInfo.Order}})  {{else}} {{if $fieldValue.IsIndex}}@index(group: "{{$fieldValue.IndexInfo.Group}}", sort: "{{$fieldValue.IndexInfo.Sort}}", order: {{$fieldValue.IndexInfo.Order}}){{end}}{{end}} {{if $fieldValue.IsDefault}}@default(value: {{$fieldValue.Default}}){{end}} {{if $fieldValue.IsForeign}}@foreign(table: {{$fieldValue.JointTable.Table}}, field: {{$fieldValue.JointTable.To}},{{if $fieldValue.JointTable.OnDelete=="CASCADE"}}OnDelete:{{$fieldValue.JointTable.OnDelete}}){{end}}{{end}}{{end}}
```
Comment (YourTechBud) [inline gateway/modules/schema/template.go]:
```suggestion
	{{$fieldName}}: {{if eq $fieldValue.Kind "Object"}}{{$fieldValue.JointTable.Table}}{{else}}{{$fieldValue.Kind}}{{end}}{{if $fieldValue.IsFieldTypeRequired}}!{{end}} {{if $fieldValue.IsPrimary}}@primary{{end}} {{if $fieldValue.IsUnique}}@unique(group: "{{$fieldValue.IndexInfo.Group}}", order: {{$fieldValue.IndexInfo.Order}})  {{else}} {{if $fieldValue.IsIndex}}@index(group: "{{$fieldValue.IndexInfo.Group}}", sort: "{{$fieldValue.IndexInfo.Sort}}", order: {{$fieldValue.IndexInfo.Order}}){{end}}{{end}} {{if $fieldValue.IsDefault}}@default(value: {{$fieldValue.Default}}){{end}} {{if $fieldValue.IsForeign}}@foreign(table: {{$fieldValue.JointTable.Table}}, field: {{$fieldValue.JointTable.To}},{{if eq $fieldValue.JointTable.OnDelete "cascade"}} onDelete:{{$fieldValue.JointTable.OnDelete}}){{end}}{{end}}{{end}}
```

File: gateway/modules/schema/schema.go [hunk 6]
```diff
@@ -236,6 +237,13 @@ func getCollectionSchema(doc *ast.Document, dbName, collectionName string) (mode
 							case "field", "to":
 								val, _ := utils.ParseGraphqlValue(arg.Value, nil)
 								fieldTypeStuct.JointTable.To = val.(string)
+
+							case "ondelete":
```
Comment (YourTechBud) [inline gateway/modules/schema/schema.go]:
```suggestion
							case "onDelete":
```

**Comment by YourTechBud** [comment]
@sharadregoti make sure no query is fired when schema containing foreign key isn't changed

**Comment by sharadregoti** [comment]
@YourTechBud tested this pr. Everything works fine. No queries are fired when schema changes.
but reload-scheam or inspect schema throws error as there is an error in schema template string.

**Comment by sharadregoti** [comment]
@YourTechBud for confirmation the get schema context you asked to check is reload-schema or inspect schema correct ?

**Comment by YourTechBud** [comment]
@sharadregoti yes

**Pull request merged**